src/cdk/v2/destinations/zoho/utils.ts

```diff
@@ -226,13 +339,377 @@ const searchRecordIdV2 = async ({
     return result;
   } catch (error: any) {
     return {
-      erroneous: true,
+      status: false,
       message: error.message,
     };
   }
 };
 
-// ref : https://www.zoho.com/crm/developer/docs/api/v6/upsert-records.html#:~:text=The%20trigger%20input%20can%20be%20workflow%2C%20approval%2C%20or%20blueprint.%20If%20the%20trigger%20is%20not%20mentioned%2C%20the%20workflows%2C%20approvals%20and%20blueprints%20related%20to%20the%20API%20will%20get%20executed.%20Enter%20the%20trigger%20value%20as%20%5B%5D%20to%20not%20execute%20the%20workflows.
+// ============================================================================
+// Batched Deletion with IN/OR + Post-Filter Approach
+// ============================================================================
+
+/**
+ * Determines if batched deletion lookup is enabled for a given workspace.
+ * Uses ZOHO_DELETION_BATCHING_LOOKUP environment variable as a feature flag.
+ * If env var is not set or empty, batching is enabled by default.
+ * If env var is set, only workspaces in the comma-separated list get batching.
+ *
+ * @param {string} workspaceId - The workspace ID to check
+ * @returns {boolean} True if batched deletion lookup should be used, false for legacy one-by-one approach
+ *
+ * @example
+ * // Env: ZOHO_DELETION_BATCHING_LOOKUP="workspace1,workspace2"
+ * isDeletionLookupBatchingEnabled('workspace1') // Returns: true
+ * isDeletionLookupBatchingEnabled('workspace3') // Returns: false
+ *
+ */
+const isDeletionLookupBatchingEnabled = (workspaceId: string | undefined) => {
+  // If no workspaceId provided (tests or legacy behavior), disable batching
+  if (!workspaceId) {
+    return false;
+  }
+
+  // Check if this workspace is in the feature flag list
+  const zohoDeletionBatchingLookup = process.env.ZOHO_DELETION_BATCHING_LOOKUP?.split(',')?.map?.(
```

Comment:
DEST_<dest_type>_<purpose/feature>_WORKSPACE_IDS
```suggestion
  const zohoDeletionBatchingLookup = process.env.DEST_ZOHO_DELETION_BATCHING_SUPPORTED_WORKSPACE_IDS?.split(',')?.map?.(
```

Reply:
Also, we don't have to process this env for each request. Let's compute the IDs as part of startup and re-use them.